tests/test_contracts/test_class_contracts.py

```diff
@@ -424,5 +424,40 @@ def test_no_premature_check_from_deep_helper_in_init() -> None:
     assert dark_widget.secondary_color == "mahogany"
 
 
+def test_invariant_with_function_defined_in_module() -> None:
+    """Test that a representation invariant violation is detected when the invariant
+    contains a call to a function (top-level) defined by the user
+    This test is based on the code found at ./test_nested_preconditions_example.py
+    """
+    import test_nested_preconditions_example as example
```

Comment:
Move this import statement to the top level of this file. Also, just to be safe don't name that module starting with `test_`, we don't want pytest to scan the file.